docs: clarify release notes verification steps (#4960)
mongodb/node-mongodb-native: etc/notes/releasing.md

    - You may skip this step if you are releasing a package other than the driver.
1. Comment "`run release_notes`" on the release PR.
    - This will kick off the action that reads the notes from each PR going into the release.
    - Double check the result looks logically organized and complete
        - As a rule of thumb, every `feat` and `fix` should have a corresponding release note
        - Cross-reference the list of tickets in the corresponding jira release for consistency

>     - You may edit the PR body for any quick edits or re-orderings.

Author: ```suggestion
    - You may modify the release PR body directly for any quick edits or re-orderings.
```

>     - If there are a number of changes, edit the original PRs and post a new `run` comment

Author: ```suggestion
    - If there are any non-cosmetic changes, edit the original PRs and post a new `run release_notes` comment
```
